src/test/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/KNNScriptScoringIT.java

```diff
@@ -308,4 +322,169 @@ public void testKNNScoreforNonVectorDocument() throws Exception {
         assertEquals(0.33333, scores.get(0), 0.001);
         assertEquals(Float.MIN_VALUE, scores.get(1), 0.001);
     }
+
+    @SuppressWarnings("unchecked")
+    public void testHammingScriptScore_Long() throws Exception {
```

Comment:
How about we add a testcase in such a way a segment contains docs with field name and with out the field name.   This would test how the scoring is handled for the segments with the above combination.  

We could try forcemerge and then assert the docs with out the similarity field are not in the top hits. One test for long and one test for base64 should be good?

Reply:
Sure, so at the moment, I add 4 docs with the field and then 3 docs without the field, and then confirm the docs without the field arent in the results. What I will do is put in a refresh in between the addition of the docs with the field and the docs without the field, to confirm that a segment either has only docs with the field or only docs without the field. Then, IO will confirm the docs without the field arent in the results. Then I will forcemerge to 1 segment, and repeat the experiment.